### PR TITLE
Issue #5722: Update DTD public names

### PR DESCRIPTION
**Pull request by subkrish**
Issue #5722 

All DTD names have been updated in the required `.xml`, `.dtd`, `.java` and `.xml.vm` files.

### CODE REVIEW

**Comment by subkrish** [comment]
@romani 

I have updated the remaining items as well. Done.

**Comment by rnveach** [comment]
@romani
> more items to update:
> src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlLoader.java

Are we still supporting the old DTD names or only the new ones?
If we are supporting both, shouldn't the new DTD names be in a new DTD version?
If we aren't supporting both, doesn't this mean old versions won't work anymore and this is a backward breaking change?

File: src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java [hunk 1]
```diff
@@ -73,31 +73,31 @@
 
     /** The public ID for version 1_0 of the configuration dtd. */
     private static final String DTD_PUBLIC_ID_1_0 =
-        "-//Puppy Crawl//DTD Check Configuration 1.0//EN";
+        "-//Checkstyle//DTD Checkstyle Configuration 1.0//EN";
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java]: See comment at https://github.com/checkstyle/checkstyle/pull/5869#issuecomment-392934305 .
If we are dropping old name, then this means we won't support old DTD names.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java]: good catch.

I updated issue description with "attention" note and explanation.

@subkrish , we need to keep DTD_PUBLIC_ID_1_0 unchanged and all other variable in this class.
Please create new variables DTD_PUBLIC_CS_ID_X_X with new values and add new keys for same value at method `createIdToResourceNameMap` of `ConfigurationLoader`.

The same should be done in other classes `*Loader`where "Puppy Crawl" should stay and new filed with proper value be added:
```
$ grep -Rl ".dtd\";" --include="*.java" *
src/main/java/com/puppycrawl/tools/checkstyle/XmlLoader.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlLoader.java
src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionsLoader.java
src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java
src/main/java/com/puppycrawl/tools/checkstyle/PackageNamesLoader.java
```

**Comment by romani** [comment]
@subkrish ,

> What should I add as comment for the new variables?

you can copy comment from old variables

**Comment by romani** [comment]
@subkrish , let me know if there is more problem.

**Comment by romani** [comment]
@subkrish , ping.

**Comment by subkrish** [comment]
@romani 

Sorry for the delay.

I have updated as requested.

**Comment by rnveach** [comment]
@subkrish Code coverage is damaged.
> class com.puppycrawl.tools.checkstyle.XmlLoader: lines covered ratio is 0.89, but expected minimum is 1.00

**Comment by romani** [comment]
@subkrish , please rebase and fix more cases:
```
$ grep -Rl "Puppy Crawl" --exclude-dir="target" --exclude="*Loader.java" *
src/it/java/org/checkstyle/suppressionxpathfilter/AbstractXpathRegressionTest.java
src/it/resources/org/checkstyle/suppressionxpathfilter/importcontrol/SuppressionXpathRegressionImportControlFour.xml
src/it/resources/org/checkstyle/suppressionxpathfilter/importcontrol/SuppressionXpathRegressionImportControlOne.xml
src/it/resources/org/checkstyle/suppressionxpathfilter/importcontrol/SuppressionXpathRegressionImportControlTwo.xml
```

I relaunched appveyor build, I can not reproduce coverage problem on my local (linux), travis is green too.
Appveyor is still failing, looks like OS specific issue.

@rnveach , we need your help here to share report from windows OS.

**Comment by rnveach** [comment]
@romani I have already been helping @subkrish with the issue in gitter today.
The issue is `PackageNamesLoader`'s constructor is not used anymore and should be removed. I don't know why coverage is saying it is an OS specific issue.

**Comment by romani** [comment]
on linux (base on https://github.com/checkstyle/checkstyle/wiki/How-to-generate-UT-coverage-report):
image

**Comment by subkrish** [comment]
@romani @rnveach 

I generated the JaCoCo report on my MacBook Pro and I was able to see loss of coverage of the first constructor in `XMLLoader` because of the change I made in `PackageNamesLoader`. By removing those lines coverage issue will be solved as no other class actually uses it.

> please rebase and fix more cases:

@romani 
I will update these files as well and push once you confirm if the constructor can be removed or not.

**Comment by romani** [comment]
Please remove unused code

**Comment by subkrish** [comment]
@romani @rnveach 

Rebased and updated. I removed unnecessary code and made changes to the files in [https://github.com/checkstyle/checkstyle/pull/5869#issuecomment-395639897](url).

Done.

**Comment by subkrish** [comment]
@romani

The TeamCity build failure is from master and is not from my end.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlLoader.java [hunk 2]
```diff
@@ -46,22 +46,42 @@
...
         "-//Puppy Crawl//DTD Import Control 1.2//EN";
 
+    /** The new public ID for version 1_2 of the configuration dtd. */
+    private static final String DTD_PUBLIC_CS_ID_1_2 =
+        "-//Checkstyle//DTD ImportControl Configuration 1.22//EN";
+
     /** The public ID for the configuration dtd. */
     private static final String DTD_PUBLIC_ID_1_3 =
-            "-//Puppy Crawl//DTD Import Control 1.3//EN";
+        "-//Puppy Crawl//DTD Import Control 1.3//EN";
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlLoader.java]: @subkrish There are alot of unnecessary changes like this.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlLoader.java [hunk 3]
```diff
@@ -76,7 +96,7 @@
 
     /** The resource for the configuration dtd. */
     private static final String DTD_RESOURCE_NAME_1_3 =
-            "com/puppycrawl/tools/checkstyle/checks/imports/import_control_1_3.dtd";
+        "com/puppycrawl/tools/checkstyle/checks/imports/import_control_1_3.dtd";
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlLoader.java]: Here.

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionsLoader.java [hunk 4]
```diff
@@ -50,33 +50,48 @@
...
+    /** The new public ID for version 1_2 configuration dtd. */
+    private static final String DTD_PUBLIC_CS_ID_1_2 =
+        "-//Checkstyle//DTD SuppressionFilter Configuration 1.2//EN";
     /** The resource for the configuration dtd. */
     private static final String DTD_SUPPRESSIONS_NAME_1_2 =
         "com/puppycrawl/tools/checkstyle/suppressions_1_2.dtd";
     /** The public ID for the configuration dtd. */
     private static final String DTD_PUBLIC_ID_1_1_XPATH =
-            "-//Puppy Crawl//DTD Suppressions Xpath Experimental 1.1//EN";
+        "-//Puppy Crawl//DTD Suppressions Xpath Experimental 1.1//EN";
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionsLoader.java]: here.

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionsLoader.java [hunk 5]
```diff
@@ -50,33 +50,48 @@
...
+    /** The new public ID for version 1_1 configuration dtd. */
+    private static final String DTD_PUBLIC_CS_ID_1_1_XPATH =
+        "-//Checkstyle//DTD SuppressionXpathFilter Experimental Configuration 1.1//EN";
     /** The resource for the configuration dtd. */
     private static final String DTD_SUPPRESSIONS_NAME_1_1_XPATH =
-            "com/puppycrawl/tools/checkstyle/suppressions_1_1_xpath_experimental.dtd";
+        "com/puppycrawl/tools/checkstyle/suppressions_1_1_xpath_experimental.dtd";
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionsLoader.java]: here.

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionsLoader.java [hunk 6]
```diff
@@ -50,33 +50,48 @@
...
     /** The public ID for the configuration dtd. */
     private static final String DTD_PUBLIC_ID_1_2_XPATH =
-            "-//Puppy Crawl//DTD Suppressions Xpath Experimental 1.2//EN";
+        "-//Puppy Crawl//DTD Suppressions Xpath Experimental 1.2//EN";
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionsLoader.java]: here.

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionsLoader.java [hunk 7]
```diff
@@ -50,33 +50,48 @@
...
+    /** The new public ID for version 1_2 configuration dtd. */
+    private static final String DTD_PUBLIC_CS_ID_1_2_XPATH =
+        "-//Checkstyle//DTD SuppressionXpathFilter Experimental Configuration 1.2//EN";
     /** The resource for the configuration dtd. */
     private static final String DTD_SUPPRESSIONS_NAME_1_2_XPATH =
-            "com/puppycrawl/tools/checkstyle/suppressions_1_2_xpath_experimental.dtd";
+        "com/puppycrawl/tools/checkstyle/suppressions_1_2_xpath_experimental.dtd";
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionsLoader.java]: here.

**Comment by rnveach** [comment]
I don't see anything else wrong.

**Comment by rnveach** [comment]
I re-approved because it was pointed out some of the new indentations aligned the entire file.
